simplest implementation of promisc flag on open_link, works on Linux

gabrik opened the pull request:
Simple implementation of the promisc parameter on open_link
to enable promiscuos mode on a interface while opening it, tested on 
Ubuntu 16.04.5
Ocaml 4.07.1
dune 1.6.3

See https://github.com/haesbaert/rawlink/issues/12

Signed-off-by: gabrik <gabriele.baldoni@gmail.com>

haesbaert commented on lib/rawlink_stubs.c:
```diff
@@ -271,40 +264,58 @@ af_packet_setfilter(int fd, value vfilter)
+
+	strncpy(ifopts.ifr_name, ifname, IFNAMSIZ - 1);
+	caml_enter_blocking_section();
+	r = ioctl(fd, SIOCGIFFLAGS, &ifopts);
```
You have to test the error of each syscall and abort, also ifopts should be bzroed before.

haesbaert commented on lib/rawlink_stubs.c:
```diff
@@ -271,40 +264,58 @@ af_packet_setfilter(int fd, value vfilter)
+int af_packet_set_promisc(int fd, int promisc, char *ifname)
+{
+	struct ifreq ifopts;
+	int r;
```
new line after declarations, also keep same indentation style.

hannesm commented on lib/rawlink.ml:
```diff
@@ -35,16 +35,16 @@ type driver =
-let open_link ?filter ifname =
-  { fd = opensock ?filter:filter ifname;
+let open_link ?promisc ?filter ifname =
+  { fd = opensock ?promisc:promisc ?filter:filter ifname;
```
FWIW, you can drop `:promisc` and `:filter` here:
```OCaml
let fd = opensock ?promisc ?filter ifname in
{ fd ; packets = ref [] ; buffer = Cstruct.create 65536 }
```

gabrik commented:
Hi both, I have updated the PR addressing the comments, the indentation style should be the right one, so feel free to borrow this last version for the feature branch, also I have a question on the Lwt version, why Lwt_rawlink.open_link return t instead of returing t Lwt.t ?

haesbaert commented:
I merged something inspired on this, thanks for the report.
See b7284a9aecf3da5799ac258a4407df7a3fc16447
